crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -27,38 +29,35 @@ use crate::{
 /// of provers for proof generation. Proving is currently unimplemented and is instead simulated via
 /// the given proof time and failure rate.
 pub struct BatchBuilder {
-    pub batch_interval: Duration,
-    pub workers: NonZeroUsize,
-    /// Used to simulate batch proving by sleeping for a random duration selected from this range.
-    pub simulated_proof_time: Range<Duration>,
-    /// Simulated block failure rate as a percentage.
-    ///
-    /// Note: this _must_ be sign positive and less than 1.0.
-    pub failure_rate: f32,
+    worker_pool: JoinSet<()>,
+    batch_interval: Duration,
     /// The batch prover to use.
     ///
     /// If not provided, a local batch prover is used.
     batch_prover: BatchProver,
+    /// Simulated block failure rate as a percentage.
+    ///
+    /// Note: this _must_ be sign positive and less than 1.0.
+    failure_rate: f64,
+    store: StoreClient,
 }
 
 impl BatchBuilder {
     /// Creates a new [`BatchBuilder`] with the given batch prover URL.
     ///
     /// If no URL is provided, a local batch prover is used.
-    pub fn new(batch_prover_url: Option<Url>) -> Self {
-        let batch_prover = match batch_prover_url {
-            Some(url) => BatchProver::new_remote(url),
-            None => BatchProver::new_local(MIN_PROOF_SECURITY_LEVEL),
-        };
+    pub fn new(store: StoreClient, workers: NonZeroUsize, batch_prover_url: Option<Url>) -> Self {
```

Comment:
nit: I would rename `workers` to `num_workers`.

Also, maybe it is worth updating the doc comments for this function.